couchbase3/src/main/scala/org/apache/pekko/stream/connectors/couchbase3/scaladsl/CouchbaseSource.scala

```diff
@@ -0,0 +1,226 @@
+   */
+  def getType[T](id: String, target: TypeRef[T], options: GetOptions = GetOptions.getOptions)(
+      implicit asyncCollection: AsyncCollection): Source[T, NotUsed] = {
+    get(id, options).map(_.contentAs(target))
+  }
+
+  /**
+   * same to Get option, but reads from all replicas on the active node
+   * @param options reference to Couchbase options doc
+   * @see [[CouchbaseSource#get]]
+   */
+  def getAllReplicas(id: String, options: GetAllReplicasOptions = GetAllReplicasOptions.getAllReplicasOptions)(
+      implicit asyncCollection: AsyncCollection): Source[GetReplicaResult, NotUsed] =
+    Source
+      .completionStage(asyncCollection.getAllReplicas(id, options))
+      .flatMapConcat { res =>
+        Source.fromJavaStream(() => res.stream())
+          .flatMapConcat(Source.completionStage)
+      }
+
+  /**
+   * reference to [[CouchbaseSource.getAllReplicas]], deserialize to Couchbase JsonObject
+   */
+  def getAllReplicasJson(id: String, options: GetAllReplicasOptions = GetAllReplicasOptions.getAllReplicasOptions)(
+      implicit asyncCollection: AsyncCollection): Source[JsonObject, NotUsed] =
+    getAllReplicas(id, options)
+      .map(_.contentAsObject())
+
+  /**
+   * reference to [[CouchbaseSource.getAllReplicas]], deserialize to class
+   * If you add DefaultScalaModule to jackson of couchbase, it could deserialize to scala class
+   */
+  def getAllReplicasObject[T](id: String, target: Class[T],
+      options: GetAllReplicasOptions = GetAllReplicasOptions.getAllReplicasOptions)(
+      implicit asyncCollection: AsyncCollection): Source[T, NotUsed] =
+    getAllReplicas(id, options)
+      .map(_.contentAs(target))
+
+  /**
+   * reference to [[CouchbaseSource.getAllReplicasObject]], deserialize to class with Generics
+   */
+  def getAllReplicasType[T](id: String, target: TypeRef[T],
+      options: GetAllReplicasOptions = GetAllReplicasOptions.getAllReplicasOptions)(
+      implicit asyncCollection: AsyncCollection): Source[T, NotUsed] =
+    getAllReplicas(id, options)
+      .map(_.contentAs(target))
+
+  /**
+   * like Get[[CouchbaseSource.get]], batch get documents from collection by ScanType[[ScanType]]
+   */
+  def scan(scanType: ScanType, options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[ScanResult, NotUsed] =
+    Source
+      .completionStage(asyncCollection.scan(scanType, options))
+      .flatMapConcat(res => Source.fromJavaStream(() => res.stream()))
+
+  /**
+   * reference to [[CouchbaseSource.scan]], deserialize to Couchbase JsonObject
+   */
+  def scanJson(scanType: ScanType, options: ScanOptions = ScanOptions.scanOptions())(
+      implicit asyncCollection: AsyncCollection): Source[JsonObject, NotUsed] =
+    scan(scanType, options)
+      .map(_.contentAsObject())
+
+  /**
+   * reference to [[CouchbaseSource.scan]], deserialize to class
+   * If you add DefaultScalaModule to jackson of couchbase, it could deserialize to scala class
```

Comment:
what does this bit about Jackson mean? How can users add this module?

Reply:
like this, when user want to create a cluser(connection), use need to set environment with serializer.
```
  private val jacksonMapper = JsonMapper.builder()
    .addModule(DefaultScalaModule)
    .build()
  private val serializer: JacksonJsonSerializer = JacksonJsonSerializer.create(jacksonMapper)
  private val environment = ClusterEnvironment.builder().jsonSerializer(serializer).build()
  lazy val cluster: Cluster = Cluster.connect(connectionString,
    ClusterOptions.clusterOptions(username, password).environment(environment))
```

Coucase's default Jackson does not support scala， And I have a new idea, add a default cluster connection , the default connection will automatically apply this, and then we can remove the description here， besides, user also can set theri owner cluster by coding.

Reply:
It would be great to automatically add the Default Scala module but we need to let users add more modules. We can just document how to set up Jackson as you have described above if it is too awkward to come up with our own wrapper.

Reply:
I have updated the description